specification/sbi_atee.adoc

```diff
@@ -10,8 +39,1238 @@ shown in the figure above, the TEE Runtime (operating in VS-mode) supports
+parameters in the `tvm_aia_params` structure at the non-confidential physical address
+at `tvm_aia_params_addr`. The `tvm_aia_params_len` is the byte-length of the `tvm_aia_params` 
+structure.
+
+This cannot be called after `sbi_tee_host_finalize_tvm()`.
+
+The format and semantics of the `tvm_aia_params_addr` structure appears below.
+
+[source, C]
+-------
+struct tvm_aia_params {
+    /* 
+     * The base address of the virtualized IMSIC in TVM physical address space.
+     *
+     * IMSIC addresses follow the below pattern:
+     *
+     * XLEN-1 >=24 12 0 | | | |
+     *
+     * |xxxxxx|Group Index|xxxxxxxxxxx|Hart Index|Guest Index| 0 |
+     *
+     * The base address is the address of the IMSIC with group ID, hart ID, and guest ID of 0.
+     */
+    unsigned long imsic_base_addr;
+    /* The number of group index bits in an IMSIC address. */
+    unsigned int group_index_bits;
+    /* The location of the group index in an IMSIC address. Must be >= 24. */
+    unsigned int group_index_shift;
+    /* The number of hart index bits in an IMSIC address. */
+    unsigned int hart_index_bits;
+    /* The number of guest index bits in an IMSIC address. Must be >= log2(guests_per_hart + 1). */
+    unsigned int guest_index_bits;
+    /*
+     * The number of guest interrupt files to be implemented per VCPU. Implementations may reject
+     * configurations with guests_per_hart > 0 if nested IMSIC virtualization is not supported.
+     */
+    unsigned int guests_per_hart;
+};
+-------
+
+The possible error codes returned in `sbiret.error` are shown below.
+
+[#table_sbi_tee_interrupt_init_tvm_aia_errors]
+.TEE Interrupt Init TVM AIA
+[cols="2,3", width=90%, align="center", options="header"]
+|===
+| Error code              | Description
+| SBI_SUCCESS             | The operation completed successfully.
+| SBI_ERR_INVALID_ADDRESS | `tvm_aia_params_addr` was invalid.
+| SBI_ERR_INVALID_PARAM   | `tvm_guest_id` or `tvm_aia_params_addr` were invalid,
+                            or the TVM wasn't in the "initializing" state.
+| SBI_ERR_FAILED          | The operation failed for unknown reasons.
+|===
+
+=== Function: TEE Interrupt Set TVM AIA CPU IMSIC Addr (FID #1)
+[source, C]
+-------
+struct sbiret sbi_tee_interrupt_set_tvm_aia_cpu_imsic_addr(unsigned long tvm_guest_id,
+                                                           unsigned long tvm_vcpu_id,
+                                                           unsigned long tvm_vcpu_imsic_gpa);
+-------
+
+Sets the guest physical address of the specified VCPU’s virtualized IMSIC to `tvm_vcpu_imsic_gpa`.
+The `tvm_vcpu_imsic_gpa` must be valid for the AIA configuration that was set by 
+`sbi_tee_interrupt_init_tvm_aia()`. No two VCPUs may share the same `tvm_vcpu_imsic_gpa`.
+
+This can be called only after `sbi_tee_interrupt_init_tvm_aia()` and before `sbi_tee_host_finalize_tvm()`.
+All VCPUs in an AIA-enabled TVM must their IMSIC configuration set prior to calling
```

Comment:
missing word = All VCPUs in an AIA-enabled TVM must *have* their IMSIC configuration set prior...